torch/quantization/fx/quantize.py

```diff
@@ -725,22 +718,44 @@ def is_observed(input_arg):
                     observed.add(node.name)
         observed_graph.output(load_arg(input_graph.result))
 
-        return GraphModule(input_root, observed_graph)
+        observed = GraphModule(input_root, observed_graph)
+        self.save_state(observed)
+        return observed
+
+    def save_state(self, observed):
+        observed._activation_post_process_map = self.activation_post_process_map
+        observed._patterns = self.patterns
+        observed._qconfig_map = self.qconfig_map
+
+    def restore_state(self, observed):
+        err_msg = 'please make sure the model is produced by prepare'
+        assert hasattr(observed, '_activation_post_process_map'), 'did not found ' + \
+            '_activation_post_process attribute ' + err_msg
+        assert hasattr(observed, '_patterns'), 'did not found ' + \
+            '_patterns attribute ' + err_msg
+        assert hasattr(observed, '_qconfig_map'), 'did not found ' + \
+            '_qconfig_map attribute ' + err_msg
+        self.activation_post_process_map = observed._activation_post_process_map
+        self.patterns = observed._patterns
+        self.qconfig_map = observed._qconfig_map
```

Comment:
do we know how long prepare and convert steps take for real models?  If performance is good, it might be worth removing the saving of this state and instead regenerate it dynamically, to reduce complexity.

Reply:
seems like
* `self.activation_post_process_map` can be determined by looking at the graph nodes
* `self.patterns` is just a call to `get_quant_patterns` or `get_dynamic_quant_patterns`, so we can regenerate it as long as we know the quantization type
* `self.qconfig_map` seems to only depend on the attached qconfig

can we just regenerate this all dynamically?  It will be cleaner.

Reply:
prepare and convert should be fast. sure I think we can generate these

Reply:
cool, thx! happy to help if needed, lmk.  I'd expect this to not be a p0, but probably more important as the codebase matures and there are more people reading the code, doing custom things, etc

Reply:
btw, might be n00b questions,
* for `self.activation_post_process_map`, is there an O(1) lookup API to look up a node by name from `fx.Graph`?  Seems like that would be useful, and then we could delete the custom one
* for `self.qconfig_map`, is the state ever modified after being generated?  If not, could the state be removed altogether by just always looking up the qconfig via the specified rules (which should be fast, and removing state will simplify things)